src/java.base/share/classes/java/lang/constant/ReferenceClassDescImpl.java

```diff
@@ -55,6 +55,11 @@ final class ReferenceClassDescImpl implements ClassDesc {
         this.descriptor = descriptor;
     }
 
+    @Override
+    public String internalName() {
+        return isArray() ? descriptorString() : descriptorString().substring(1, descriptorString().length() - 1);
```

Comment:
I suggest to cache the internal name to avoid repeated substring-ing with each call.
`ClassDesc` instances are frequently used as static constants serving for generations or transformations of many classes, methods, instructions. 
It is highly expected that if this method is invoked at least once on the particular instance, it will be invoked many times on the same instance.